lms/djangoapps/lti_provider/views.py

```diff
@@ -58,8 +61,18 @@ def lti_launch(request, course_id, usage_id):
     # Check the LTI parameters, and return 400 if any required parameters are
     # missing
     params = get_required_parameters(request.POST)
+    
+    return_url = request.POST.get('launch_presentation_return_url')
+    template400 = Template(render_to_string('static_templates/400.html', {'return_url': return_url}))
     if not params:
-        return HttpResponseBadRequest()
+        return HttpResponseBadRequest(template400.render())
+
+    if params['lti_version'] != 'LTI-1p0':
```

Comment:
the same here: `params['lti_version'] != 'LTI-1p0'` and `params['lti_message_type'] != 'basic-lti-launch-request'`
So strict rules - do we really need them? I'm not sure that all existing external LMS pass these params through the LTI.

// cc @dpowell